src/cmd/sanity_test/sanity_tests.go

```diff
@@ -105,260 +126,328 @@ func checkEmailRestoration(
 
 		if childFolderCount > 0 {
-			parentFolder := fullFolderName
-
-			checkAllSubFolder(client, testUser, child, parentFolder, messageCount)
+			checkAllSubFolder(ctx, client, testUser, child, fullFolderName, messageCount)
 		}
 	}
 }
 
-func checkOnedriveRestoration(client *msgraphsdk.GraphServiceClient, testUser, folderName string, startTime time.Time) {
-	file := make(map[string]int64)
-	folderPermission := make(map[string][]string)
-	restoreFolderID := ""
+func checkOnedriveRestoration(
+	ctx context.Context,
+	client *msgraphsdk.GraphServiceClient,
+	testUser, folderName string,
+	startTime time.Time,
+) {
+	var (
+		// map itemID -> item size
+		file = make(map[string]int64)
+		// map itemID -> permission id -> []permission roles
+		folderPermission = make(map[string]map[string][]string)
+		folderID         = ""
+	)
 
-	drive, err := client.UsersById(testUser).Drive().Get(context.Background(), nil)
+	drive, err := client.
+		UsersById(testUser).
+		Drive().
+		Get(ctx, nil)
 	if err != nil {
-		fmt.Printf("Error getting the drive: %v\n", err)
-		os.Exit(1)
+		fatal(ctx, "getting the drive:", err)
 	}
 
-	response, err := client.DrivesById(*drive.GetId()).Root().Children().Get(context.Background(), nil)
+	var (
+		driveID   = ptr.Val(drive.GetId())
+		driveName = ptr.Val(drive.GetName())
+	)
+
+	ctx = clues.Add(ctx, "drive_id", driveID, "drive_name", driveName)
+
+	response, err := client.
+		DrivesById(driveID).
+		Root().
+		Children().
+		Get(ctx, nil)
 	if err != nil {
-		fmt.Printf("Error getting drive by id: %v\n", err)
-		os.Exit(1)
+		fatal(ctx, "getting drive by id", err)
 	}
 
 	for _, driveItem := range response.GetValue() {
-		if *driveItem.GetName() == folderName {
-			restoreFolderID = *driveItem.GetId()
+		var (
+			itemID   = ptr.Val(driveItem.GetId())
+			itemName = ptr.Val(driveItem.GetName())
+			ictx     = clues.Add(ctx, "item_id", itemID, "item_name", itemName)
+		)
+
+		if itemName == folderName {
+			folderID = itemID
 			continue
 		}
 
-		var rStartTime time.Time
+		if !startTime.IsZero() {
+			folderTime, err := common.ExtractTime(itemName)
+			if err != nil && !errors.Is(err, common.ErrNoTimeString) {
+				fatal(ictx, "extracting time from file name", err)
+			}
+
+			if !errors.Is(err, common.ErrNoTimeString) && startTime.Before(folderTime) {
+				logger.Ctx(ictx).
+					With("folder_time", folderTime).
+					Info("skipping restore folder: not within time bound")
 
-		restoreStartTime := strings.SplitAfter(*driveItem.GetName(), "Corso_Restore_")
-		if len(restoreStartTime) > 1 {
-			rStartTime, _ = time.Parse(time.RFC822, restoreStartTime[1])
-			if startTime.Before(rStartTime) {
-				fmt.Printf("The restore folder %s was created after %s. Will skip check.", *driveItem.GetName(), folderName)
 				continue
 			}
 		}
 
 		// if it's a file check the size
 		if driveItem.GetFile() != nil {
-			file[*driveItem.GetName()] = *driveItem.GetSize()
+			file[itemName] = ptr.Val(driveItem.GetSize())
```

Comment:
I'm not sure either.  @neha-Gupta1 probably needs investigation in a follow-up.

Reply:
With what I understood from graph API - all the folders were existing in `root` folder. And so the implementation.
But since the test data is getting too huge and due concurrent test runs sanity tests break. So I will be creating custom data and validate only that in coming PR.